spec/system/composer/discard_draft_spec.rb

```diff
@@ -1,68 +1,72 @@
 # frozen_string_literal: true
 
 describe "Composer - discard draft modal", type: :system do
-  fab!(:current_user) { Fabricate(:admin) }
-  fab!(:topic_1) { Fabricate(:topic) }
+  fab!(:topic)
+  fab!(:user) { Fabricate(:admin) }
 
   let(:topic_page) { PageObjects::Pages::Topic.new }
-  let(:discard_draft_modal) { PageObjects::Modals::DiscardDraft.new }
   let(:composer) { PageObjects::Components::Composer.new }
+  let(:discard_draft_modal) { PageObjects::Modals::DiscardDraft.new }
 
-  before { sign_in(current_user) }
+  before { sign_in(user) }
 
   context "when editing different post" do
-    fab!(:post_1) { Fabricate(:post, topic: topic_1, user: current_user) }
-    fab!(:post_2) { Fabricate(:post, topic: topic_1, user: current_user) }
+    fab!(:post_1) { Fabricate(:post, topic:, user:) }
+    fab!(:post_2) { Fabricate(:post, topic:, user:) }
 
-    it "shows the discard modal" do
+    it "shows the discard modal when there are changes" do
       topic_page.visit_topic(post_1.topic)
       topic_page.click_post_action_button(post_1, :edit)
-
       composer.fill_content("a b c d e f g")
       composer.minimize
+
       topic_page.click_post_action_button(post_2, :edit)
 
       expect(discard_draft_modal).to be_open
     end
+
+    it "doesn't show the discard modal when there's no changes" do
+      topic_page.visit_topic(post_1.topic)
+      topic_page.click_post_action_button(post_1, :edit)
+      composer.minimize
+
+      topic_page.click_post_action_button(post_2, :edit)
+
+      expect(discard_draft_modal).to be_closed
+      expect(composer).to be_opened
+    end
   end
 
-  context "when re-editing the first post" do
-    fab!(:post_1) { Fabricate(:post, topic: topic_1, user: current_user) }
+  context "when editing the same post" do
+    fab!(:post_1) { Fabricate(:post, topic:, user:) }
 
-    it "doesn’t show the discard modal" do
+    it "doesn’t show the discard modal even if there are changes" do
```

Comment:
The "changes" here are not changes made by the user, but rather, it's the updated post from the edit action, right? It's a little unclear.